packages/block-editor/src/components/rich-text/index.js

```diff
@@ -146,52 +138,22 @@ export function RichTextWrapper(
 			isSelected = selectionStart.clientId === clientId;
 		}
 
-		// Disable Rich Text editing if block bindings specify that.
-		let shouldDisableEditing = false;
-		if ( blockBindings && blockName in BLOCK_BINDINGS_ALLOWED_BLOCKS ) {
-			const blockTypeAttributes = getBlockType( blockName ).attributes;
-			const { getBlockBindingsSource } = unlock(
-				select( blockEditorStore )
-			);
-			for ( const [ attribute, args ] of Object.entries(
-				blockBindings
-			) ) {
-				if (
-					blockTypeAttributes?.[ attribute ]?.source !== 'rich-text'
-				) {
-					break;
-				}
-
-				// If the source is not defined, or if its value of `lockAttributesEditing` is `true`, disable it.
-				const blockBindingsSource = getBlockBindingsSource(
-					args.source
-				);
-				if (
-					! blockBindingsSource ||
-					blockBindingsSource.lockAttributesEditing
-				) {
-					shouldDisableEditing = true;
-					break;
-				}
-			}
-		}
-
 		return {
 			selectionStart: isSelected ? selectionStart.offset : undefined,
 			selectionEnd: isSelected ? selectionEnd.offset : undefined,
 			isSelected,
-			shouldDisableEditing,
 		};
 	};
-	const { selectionStart, selectionEnd, isSelected, shouldDisableEditing } =
-		useSelect( selector, [
-			clientId,
-			identifier,
-			originalIsSelected,
-			isBlockSelected,
-		] );
+	const { selectionStart, selectionEnd, isSelected } = useSelect( selector, [
+		clientId,
+		identifier,
+		originalIsSelected,
+		isBlockSelected,
+	] );
 	const { getSelectionStart, getSelectionEnd, getBlockRootClientId } =
 		useSelect( blockEditorStore );
+	// Disable Rich Text editing if block bindings specify that.
+	const shouldDisableEditing = useShouldDisableEditing();
```

Comment:
Back then, it was discussed to include the logic moved to the `useShouldDisableEditing` hook into the initial selector: [link](https://github.com/WordPress/gutenberg/pull/58958/files#diff-bec072c72a2f961596701aef67f67575b0a3b918135a6587229a0cd79bb3db44L149-L178). It should improve performance and ensure the hooks are not called more than necessary. Is there any reason to move it outside?

Reply:
So I've done some additional testing, and it looks like setting `shouldDisableEditing` in the selector results in inconsistent behavior. You're also right [in this comment](https://github.com/WordPress/gutenberg/pull/58958#discussion_r1497386710) saying that the blur isn't necessary to get the focus to work. In a previous iteration, blurring the target allowed focus to be set to the new block, though apparently this wasn't addressing the core issue.

The real problem, I'm realizing, is that because the `shouldDisableEditing` value from the selector is inconsistent, it conflicts with the [expected behavior when setting `tabIndex`](https://github.com/WordPress/gutenberg/blob/5df6af49bb9dcf9bb8093164bb242df79aa6f49f/packages/block-editor/src/components/rich-text/index.js#L396-L401), preventing focus from being set to the new block.

Here's a video going over it in more detail:

https://github.com/WordPress/gutenberg/assets/5360536/edf42231-6ed8-4204-9830-acae75bbfa47

I've pushed another commit removing the blur logic. We can also move hook logic back to the Rich Text `index.js` — it'll work as long as the logic remains outside of the selector. Personally, I think having the hook separate is cleaner and easier to reason with, but I'm happy to take either approach.

Reply:
Thanks for the explanation. I guess there is an issue when the block is selected, but I am not able to identify the why.

> Personally, I think having the hook separate is cleaner and easier to reason with, but I'm happy to take either approach.

I don't have a strong opinion whatever you prefer.

Reply:
See my comment with potential refactoring for the new hook that would address most potential performance implications: https://github.com/WordPress/gutenberg/pull/58958#discussion_r1500520572.